src/metabase/driver.clj

```diff
@@ -735,7 +735,8 @@
                               :schemas                                true
                               :test/jvm-timezone-setting              true
                               :fingerprint                            true
-                              :upload-with-auto-pk                    true}]
+                              :upload-with-auto-pk                    true
+                              :test/dynamic-dataset-loading           true}]
```

Comment:
Thanks @appleby! Setting the default to true for `dynamic-dataset-loading` was missing.